src/App.js

```diff
@@ -9,28 +9,40 @@ import TodoListItem from './TodoListItem';
 
 
 const title = "Todo List";
-
-  function useSemiPersistentState () {
   // First you're getting the "savedTodoList" data from localStorage.
   // Then you're parsing that data (which is a JSON string)
   // Then you're passing the parsed data as the first argument to `useState`, which gives the state its default value (what `todoList` starts as)
-  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem("savedTodoList")));
+
+const App = () => {
+  const [todoList, setTodoList] = useState([]);
+  
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const promise = new Promise((resolve, reject) => {
+      
+      setTimeout(() =>{
+        resolve({data:{todoList:JSON.parse(localStorage.getItem("savedTodoList"))}})
```

Comment:
You need a default state after the JSON.parse(localStorage.getItem("savedTodoList"). If you don't have one you'll get an error when there is no savedTodoList item in local storage.